src/middlewares/Cache.spec.ts

```diff
@@ -0,0 +1,128 @@
+import {Cache, ICacheMiddlewareStore} from "./Cache";
+import {FetchResponse} from "./Fetch";
+
+// noinspection TsLint
+declare const Response: FetchResponse<string>;
+declare const localStorage: ICacheMiddlewareStore;
+
+describe("Cache", () => {
+    it("is defined", () => {
+        expect(Cache).toBeDefined();
+    });
+
+    describe("constructor()", () => {
+        it("can be instantiated", () => {
+            expect(new Cache() instanceof Cache).toBeTruthy();
+        });
+
+        it("accepts storage engine in instantiation", () => {
+            expect(new Cache(localStorage) instanceof Cache).toBeTruthy();
+        });
+    });
+
+    describe("process()", () => {
+        it("is a function", () => {
+            expect(typeof (new Cache()).process).toBe("function");
+        });
+
+        it("calls next() if cache not enabled", (done) => {
+            const cacheMw: Cache = new Cache();
+            const mock: Promise<any> = Promise.resolve(new Response("fetched response", undefined));
+            const cached = cacheMw.process({ url: "/test" }, () => {
+                setTimeout(() => done(), 20);
+
+                return mock;
```

Comment:
change this to calls next() and returns its value. if cache isn't enabled it should pass through anything. This will simplify the test as well as it will remove the need for promises.

const next = () => 42;
const result = cacheMw.process({}, next);
expect(result).toBe(42);